### PR TITLE
use formly to create the login form

### PR DESCRIPTION
**Pull request by SGrueber**
<br /><br />## PR Type<br /><br /><br /><br />[ ] Bugfix<br />[ ] Feature<br />[ ] Code style update (formatting, local variables)<br />[x] Refactoring (no functional changes, no API changes)<br />[ ] Build-related changes<br />[ ] CI-related changes<br />[ ] Documentation content changes<br />[ ] Application / infrastructure changes<br />[ ] Other: <br /><br />## What Is the Current Behavior?<br />The login form is implemented based on the Angular reactive forms module.<br /><br /><br /><br />## What Is the New Behavior?<br />The login form is created with the help of the formly module.<br /><br />## Does this PR Introduce a Breaking Change?<br /><br /><br /><br />[ ] Yes<br />[x] No<br /><br />## Other Information<br />

[AB#65083](https://dev.azure.com/intershop-com/cefd1005-00a7-4c79-927f-a16947d1b2e6/_workitems/edit/65083)

### CODE REVIEW

**Review by M-Behr — DISMISSED** [review]
Perfect 👍

File: src/app/shared/components/login/login-form/login-form.component.spec.ts [hunk 1]
```diff
@@ -21,8 +23,19 @@ describe('Login Form Component', () => {
     accountFacade = mock(AccountFacade);
 
     await TestBed.configureTestingModule({
-      declarations: [LoginFormComponent, MockComponent(ErrorMessageComponent), MockComponent(InputComponent)],
-      imports: [ReactiveFormsModule, RouterTestingModule, TranslateModule.forRoot()],
+      declarations: [LoginFormComponent, MockComponent(ErrorMessageComponent)],
+      imports: [
+        FormlyModule.forRoot({
```
Comment (MaxKless) [inline src/app/shared/components/login/login-form/login-form.component.spec.ts:28]: Don't use the FormlyTestingComponentsModule, just use the FormlyTestingModule. It includes a preconfigured FormlyModule and the ReactiveFormsModule. FormlyTestingComponentsModule is for testing types, wrappers etc

File: src/app/shared/formly/formly.module.ts [hunk 2]
```diff
@@ -41,7 +41,7 @@ import { ValidationWrapperComponent } from './wrappers/validation-wrapper/valida
     CaptchaExportsModule,
     CommonModule,
     DirectivesModule,
-    FormlyBaseModule.forRoot({
+    FormlyBaseModule.forChild({
```
Comment (MaxKless) [inline src/app/shared/formly/formly.module.ts:44]: This is not a problem. I double checked the formly source code and there should not be any problems with this :)

File: src/app/shared/formly/types/text-input-field/text-input-field.component.ts [hunk 3]
```diff
@@ -9,4 +9,19 @@ import { FieldType } from '@ngx-formly/core';
...
+  textInputFieldTypes = ['text', 'email', 'password', 'tel'];
+
+  prePopulate(field: FormlyFieldConfig) {
+    if (!field.templateOptions?.type) {
+      field.templateOptions.type = 'text';
+      return;
+    }
+
+    if (!this.textInputFieldTypes.includes(field.templateOptions.type)) {
+      throw new Error(
```
Comment (MaxKless) [inline src/app/shared/formly/types/text-input-field/text-input-field.component.ts:22]: Very nice :)

**Review by MaxKless — CHANGES_REQUESTED** [review]

**Review by iwiederhold — APPROVED** [review]
YESSSSSSS!

**Review by MaxKless — APPROVED** [review]

**Pull request merged**